CMakeLists.txt

```diff
@@ -16,495 +17,218 @@ cmake_minimum_required (VERSION 2.8.12)
+set(PLATFORM "${CMAKE_SYSTEM_NAME}")
+string(TOUPPER "${PLATFORM}" PLATFORM)
 
-# Compiler configuration
- if(NOT (("${PLATFORM}" STREQUAL "DARWIN") OR ("${PLATFORM}" STREQUAL "EXTERNAL")))
-  if(NOT CMAKE_COMPILER_IS_GNUCC)
-   message(FATAL_ERROR "gcc compiler is required")
-  endif()
- endif()
+# Remove rdynamic option
+set(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS )
+
+# Components
+set(BUILD_JERRY_CORE    ON  CACHE BOOL "Build jerry-core?")
+set(BUILD_JERRY_LIBC    ON  CACHE BOOL "Build jerry-libc?")
+set(BUILD_JERRY_LIBM    ON  CACHE BOOL "Build jerry-libm?")
+set(BUILD_JERRY_CMDLINE ON  CACHE BOOL "Build jerry command line tool?")
+set(BUILD_UNITTESTS     OFF CACHE BOOL "Build unit tests?")
+
+# Optional build settings
+set(USE_PORT_DIR              "DEFAULT"   CACHE STRING "Should we use default or external port?")
+set(USE_LIBC                  "JERRY"     CACHE STRING "should we use jerry-libc, compiler-default or external libc?")
+set(ENABLE_LTO                OFF         CACHE BOOL   "Enable LTO build?")
+set(ENABLE_ALL_IN_ONE         ON          CACHE BOOL   "Enable all-in-one build?")
+set(ENABLE_STRIP              ON          CACHE BOOL   "Discards all symbols from object files?")
+set(EXTERNAL_LIBS_INTERFACE   "UNDEFINED" CACHE STRING "Path to external libraries' include directory")
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release")
+endif()
+
+if("${PLATFORM}" STREQUAL "DARWIN")
+  set(ENABLE_LTO "OFF")
+  set(ENABLE_ALL_IN_ONE "ON")
+endif()
 
-# Imported targets prefix
- set(PREFIX_IMPORTED_LIB imported_)
+# Status messages
+message(STATUS "CMAKE_SYSTEM_NAME         " ${CMAKE_SYSTEM_NAME})
+message(STATUS "CMAKE_SYSTEM_PROCESSOR    " ${CMAKE_SYSTEM_PROCESSOR})
+message(STATUS "CMAKE_BUILD_TYPE          " ${CMAKE_BUILD_TYPE})
+message(STATUS "BUILD_JERRY_CORE          " ${BUILD_JERRY_CORE})
+message(STATUS "BUILD_JERRY_LIBC          " ${BUILD_JERRY_LIBC})
+message(STATUS "BUILD_JERRY_LIBM          " ${BUILD_JERRY_LIBM})
+message(STATUS "BUILD_JERRY_CMDLINE       " ${BUILD_JERRY_CMDLINE})
+message(STATUS "BUILD_UNITTESTS           " ${BUILD_UNITTESTS})
+message(STATUS "USE_PORT_DIR              " ${USE_PORT_DIR})
+message(STATUS "USE_LIBC                  " ${USE_LIBC})
+message(STATUS "ENABLE_LTO                " ${ENABLE_LTO})
+message(STATUS "ENABLE_ALL_IN_ONE         " ${ENABLE_ALL_IN_ONE})
+message(STATUS "ENABLE_STRIP              " ${ENABLE_STRIP})
+message(STATUS "EXTERNAL_LIBS_INTERFACE   " ${EXTERNAL_LIBS_INTERFACE})
+
+# Setup directories
+# Project binary dir
+set(PROJECT_BINARY_DIR ${CMAKE_BINARY_DIR})
+
+# Library output directory
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib/)
+
+# Executable output directory
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin/)
+
+# Archive targets output Directory
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib/)
+
+# Compile/link flags
+# build mode specific compile/link flags
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
No need for the if.
`CMAKE_C_FLAGS_RELEASE` will only be used when the build type is `Release`.